tests/api_tests/nodes/test_views.py

```diff
@@ -879,6 +879,514 @@ def test_filtering_node_with_non_bibliographic_contributor(self):
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], 'write')
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read', 'write'])
+
+        data = {
+            'permission': 'read',
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], 'read')
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read'])
+
+    def test_change_contributor_bibliographic(self):
+        data = {
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['bibliographic'], False)
+
+        self.project.reload()
+        assert_false(self.project.get_visible(self.user_two))
+
+        data = {
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)        
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['bibliographic'], True)
+
+        self.project.reload()
+        assert_true(self.project.get_visible(self.user_two))
+
+    def test_change_contributor_permission_and_bibliographic(self):
+        data = {
+            'permission': 'read',
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], 'read')
+        assert_equal(attributes['bibliographic'], False)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read'])
+        assert_false(self.project.get_visible(self.user_two))
+
+    def test_not_change_contributor(self):
+        data = {
+            'permission': None,
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], 'write')
+        assert_equal(attributes['bibliographic'], True)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read', 'write'])
+        assert_true(self.project.get_visible(self.user_two))
+
+    def test_invalid_change_inputs_contributor(self):
```

Comment:
This name does not clearly describe what the test is doing. Perhaps `test_invalid_change_to_contributor_fails`?